arrows/vxl/convert_image.cxx

```diff
@@ -74,13 +77,18 @@ combine_channels( vil_image_view< Type > const& src,
 }
 
 // ----------------------------------------------------------------------------
-// Convert a fraction of images to grey
+// Convert a fraction of images to gray
 template < typename Type >
 vil_image_view< Type >
-random_grey_conversion( vil_image_view< Type > const& src,
-                        double const random_factor )
+random_gray_conversion( vil_image_view< Type > const& src,
+                        double const random_fraction )
 {
-  if( static_cast< double >( rand() ) / RAND_MAX < random_factor )
+  std::random_device random_device;
+  std::mt19937 engine{ random_device() };
+  std::uniform_real_distribution< double > dist( 0.0, 1.0 );
+  auto random_value = dist( engine );
```

Comment:
Thanks! However, you might want to consider kicking this up to be a member of the algorithm so it is not necessary to recreate the engine and distribution for every image.

Reply:
Address, I should have refactored that into a method of the implementation class earlier.